### PR TITLE
Frontend/fix/#3438 making contact support configurable

### PR DESCRIPTION
**Pull request by MercyQueen**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

Change Log : 
- Made contact support configurable 
- Disabled tracking options for namma Yatri
- Added Fare information text in rate card
Screenshot 2023-09-29 at 6 53 20 PM

- Fetching Location name in search Location Modal only for visually impaired users.

### CODE REVIEW

File: Frontend/ui-common/src/Components/RideCompletedCard/View.purs [hunk 1]
```diff
@@ -45,21 +45,7 @@ topCardView config push =
...
-      ][  imageView
-          [ height $ V 40
-          , width $ V 40
-          , accessibility config.accessibility
-          , accessibilityHint "Contact Support : Button"
-          , imageWithFallback $ "ny_ic_headphone," <> (getCommonAssetStoreLink FunctionCall) <> "ny_ic_headphone.png"
-          , onClick push $ const Support
-          ]
-      ]
+  ][  if not config.hideContactSupport then contactSupportView config push else dummyTextView
```
Comment (Vignesh-772) [inline Frontend/ui-common/src/Components/RideCompletedCard/View.purs:48]: swap the functions instead of adding `not`

**Review by Vignesh-772 — APPROVED** [review]

File: Frontend/ui-customer/src/MerchantConfigs/DefaultConfig.purs [hunk 2]
```diff
@@ -128,4 +128,5 @@ config =
       { notify : false 
       , autoGeneratedText : ""
       }
+  , hideContactSupport : false
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/MerchantConfigs/DefaultConfig.purs:131]: change the name to `enableContactSupport` and add the default value in commonConfig for other merchants

**Review by Vignesh-772 — DISMISSED** [review]

File: Frontend/ui-customer/src/Flow.purs [hunk 3]
```diff
@@ -969,7 +972,7 @@ homeScreenFlow = do
             _ <- pure $ removeAllPolylines ""
             _ <- pure $ spy "INSIDE ELSE IF OF ONGOING" state.props.currentStage
             _ <- updateLocalStage HomeScreen
-            modifyScreenState $ HomeScreenStateType (\homeScreen -> HomeScreenData.initData{data{settingSideBar{gender = state.data.settingSideBar.gender , email = state.data.settingSideBar.email}}})
+            modifyScreenState $ HomeScreenStateType (\homeScreen -> HomeScreenData.initData{data{disability = state.data.disability, settingSideBar{gender = state.data.settingSideBar.gender , email = state.data.settingSideBar.email}}})
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Flow.purs:975]: can you move this to another function with name `resetAndUpdateDefaultsToStates` or some relevent name and do this there.

File: Frontend/ui-common/src/Components/RideCompletedCard/View.purs [hunk 4]
```diff
@@ -179,6 +146,22 @@ topCardView config push =
...
+  linearLayout
+  [ width MATCH_PARENT
+  , height WRAP_CONTENT
+  , gravity RIGHT
+  ][  imageView
+      [ height $ V 40
+      , width $ V 40
+      , accessibility config.accessibility
+      , accessibilityHint "Contact Support : Button"
+      , imageWithFallback $ "ny_ic_headphone," <> (getCommonAssetStoreLink FunctionCall) <> "ny_ic_headphone.png"
```
Comment (kavya-shree-s) [inline Frontend/ui-common/src/Components/RideCompletedCard/View.purs:160]: @MercyQueen Handle different image for different theme

File: Frontend/ui-customer/src/MerchantConfigs/DefaultConfig.purs [hunk 5]
```diff
@@ -136,4 +136,5 @@ config =
       auto : true,
       cabs : false
     }
+  , enableContactSupport : true
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/MerchantConfigs/DefaultConfig.purs:139]: @MercyQueen Can we make false in defaultConfig

**Review by kavya-shree-s — CHANGES_REQUESTED** [review]
@MercyQueen Can you also change "invoice" word to "driver receipt" in dowloaded pdf also

File: Frontend/ui-common/src/Components/RideCompletedCard/View.purs (hunk 4 above)
Comment (kavya-shree-s) [reply Frontend/ui-common/src/Components/RideCompletedCard/View.purs:160]: Its removed in the above code block , Please add it back

File: Frontend/ui-customer/src/MerchantConfigs/DefaultConfig.purs (hunk 5 above)
Comment (MercyQueen) [reply Frontend/ui-customer/src/MerchantConfigs/DefaultConfig.purs:139]: then this would be disabled for all merchants by default.

Comment (Vignesh-772) [reply Frontend/ui-customer/src/MerchantConfigs/DefaultConfig.purs:139]: add it in CommonConfig.purs

File: Frontend/ui-driver/src/MerchantConfigs/CommonConfig.purs
Comment (kavya-shree-s) [inline Frontend/ui-driver/src/MerchantConfigs/CommonConfig.purs:1]: @MercyQueen Added it in customer CommonConfig.purs

File: Frontend/android-native/mobility-customer/src/main/java/in/juspay/mobility/customer/MobilityCustomerBridge.java [hunk 6]
```diff
@@ -932,10 +933,15 @@ public void downloadPDF(String str, Context context) throws JSONException {
                 JuspayLogger.d(OTHERS, "PDF Document canvas drawn");
                 String fileNameformat;
                 String serviceName = context.getResources().getString(R.string.service);
+                String notificationHeading = context.getString(R.string.invoice_downloaded);
+                String notificationSubheading = context.getString(R.string.invoice_for_your_ride_is_downloaded);
+                System.out.println("service name ::" + serviceName);
```
Comment (kavya-shree-s) [inline Frontend/android-native/mobility-customer/src/main/java/in/juspay/mobility/customer/MobilityCustomerBridge.java:938]: Remove log line

**Review by kavya-shree-s — APPROVED** [review]

**Pull request merged**